src/test/unit/mcmc/hmc/hamiltonians/dense_e_metric_test.cpp

```diff
@@ -17,33 +18,55 @@ TEST(McmcDenseEMetric, sample_p) {
   q(0) = 5;
   q(1) = 1;
 
+  Eigen::MatrixXd  m(2,2);
+  m(0,0) = 3.0;
+  m(1,0) = -2.0;
+  m(0,1) = -2.0;
+  m(1,1) = 4.0;
+
+  Eigen::MatrixXd  m_inv(2,2);
+  double det_inv = 1.0/(3.0*4.0 - (-2.0)*(-2.0));
```

Comment:
Could you replace with this whole thing with `Eigen::MatrixXd  m_inv = m.Inverse()`?

Reply:
I used the analytic inverse to avoid numerical precision issues, which are orthogonal to what is being tested. But I can change if you prefer.